app/src/main/java/com/example/tasky/agenda/data/remote/worker/EventUploaderWorker.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.tasky.agenda.data.remote.worker
+
+import android.content.Context
+import android.net.Uri
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.example.tasky.agenda.data.remote.AgendaService
+import com.example.tasky.agenda.domain.uri.PhotoByteConverter
+import com.example.tasky.agenda.domain.uri.PhotoExtensionParser
+import com.example.tasky.core.data.Resource
+import com.example.tasky.core.data.remote.safeApiCall
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.util.UUID
+
+@HiltWorker
+class EventUploaderWorker @AssistedInject constructor(
+    @Assisted val context: Context,
+    @Assisted val workerParameters: WorkerParameters,
+    private val agendaService: AgendaService,
+    private val photoExtensionParser: PhotoExtensionParser,
+    private val photoByteConverter: PhotoByteConverter,
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
```

Comment:
I would add a retry logic here

So we try to upload the event 3 times, and then it fails

You could use runAttemptCount to check how many times the worker retried